README.md

```diff
@@ -102,8 +102,14 @@ Execution
   - [regConstraint](https://github.com/tancheng/CGRA-Mapper/blob/589fd61434966e4d4a44220b59854b1795bc7cde/test/param.json#L18): the number of registers used to temporarily hold the arrived data for later computation. Set as 8 by default.
   - [optLatency](https://github.com/tancheng/CGRA-Mapper/blob/589fd61434966e4d4a44220b59854b1795bc7cde/test/param.json#L19): used to support multi-cycle execution. If this field is not specified, every operation is done in one single-cycle. Note that there is currently no hardware support for this feature, which is supposed to be used for performance exploration only.
   - [optPipelined](https://github.com/tancheng/CGRA-Mapper/blob/589fd61434966e4d4a44220b59854b1795bc7cde/test/param.json#L23): used to enable pipelined execution of the multi-cycle operation (i.e., indicated in [optLatency](https://github.com/tancheng/CGRA-Mapper/blob/589fd61434966e4d4a44220b59854b1795bc7cde/test/param.json#L19)).
+<<<<<<< HEAD
   - [additionalFunc](https://github.com/tancheng/CGRA-Mapper/blob/589fd61434966e4d4a44220b59854b1795bc7cde/test/param.json#L24): used to enable specific functionalities on target tiles. Normally, we don't need to set this field as all the tiles already include most functionalities. By default, the `ld`/`st` is only enabled on the left most tiles. So if you wanna enable the memory access on the other tiles, this field needs to be provided. 
+ 
+=======
+  - [additionalFunc](https://github.com/tancheng/CGRA-Mapper/blob/589fd61434966e4d4a44220b59854b1795bc7cde/test/param.json#L24): used to enable specific functionalities on target tiles. Normally, we don't need to set this field as all the tiles already include most functionalities. By default, the `ld`/`st` is only enabled on the left most tiles. So if you wanna enable the memory access on the other tiles, this field needs to be provided.
+  - [incrementalMapping](https://github.com/yyan7223/CGRA-Mapper/blob/10aa217e9e995b6dfa4242e0ce121b79668e9995/test/param.json#L28C1-L28C33) `true` indicates incremental mapping while `false` indicates heuristic/exhaustive mapping. Incremental mapping re-utilizes the previous mapping results to accelerate current mapping.
```

Comment:
I am wondering why we need to let the user perform the two-step experiment. Why cannot just set this to true, and it will automatically perform heuristic/exhaustive first, then, based on the generated .json and perform the incremental mapping?

Reply:
Imagine a situation that we firstly map the **KERNEL1** with 200 DFGNodes on 400 CGRANodes, and the heuristic mapper takes 1 hour to finish this. 

Then let's say **KERNEL2**  takes 10 of 400 CGRANodes from **KERNEL1**, normally we need to re-map **KERNEL1** on the rest of 390 CGRANodes from the scratch, which will cost us 1 hour again. This is the time when we need the incremental mapping, we can simply set the **incrementalMapping** to true, then the mapping can finish within several minutes. 

But for small DFG mapping on small CGRAs, the heuristic mapper has already been fast enough and we seem do not need incremental mapping to further accelerate the mapping, so that's why I set **incrementalMapping** default to false.

 (I have not really tested the incremental mapping on large DFG and large CGRA, It will definitely bring significant mapping acceleration, but may require additional tricks to ensure mapping quality. Anyway, I think incremental mapping might be an interesting idea in CGRA just like it in FPGA)

Reply:
Thanks for the explanation. Does it mean:
 - We need to manually disable 10 nodes (before the second time mapping for `KERNEL1`) as those nodes are occupied by `KERNEL2`?
   - It is possible that the 10 nodes totally destroy the routing opportunities for the second time `KERNEL1`'s mapping? So we have to `re-map`? How is this case handled?
 - `IncrementalMapping` needs to check whether the `increMapInput.json` exists in the folder?
   - If it exists, how to guarantee it is for `KERNEL1`, instead of some random kernels?

Reply:
- I don't think we need to manually do this. There should be some sort of task scheduler that is responsible for automatically generating resource allocation decisions and corresponding `param.json` for `Kernel1` adn `Kernel2`, just like the DRIPS controller.
- Yes it is possible, but we can add a mechanism in the future like if resource changes over 50% of CGRANodes, incremental mapping is automatically disabled and start to `re-map`, just like in Vivado.
- Yes, `IncrementalMapping` should check whether `increMapInput.json` is corresponding to `Kernel1`, which can be achieved by comparing some fields in `.json` like `KernelID`. Or if developers use [CGRA-Flow](https://github.com/tancheng/CGRA-Flow), we can simply add a message box and let them choose the correct `.json`. 

Current merge request is just a initial version of incremental mapping, and I know there are lots of functions are incomplete, but I’m happy to complement all the related functions in the future and hope they actually contributes to the CGRA development rather than just some academic ideas.

Reply:
But for now, if I want to use this feature, I need manually disable some tiles (before invoking the incremental mapping) to see its effect, right?

Can you please provide the description about the constraint (users need to guarantee that `increMapInput.json` exists and the `incrementalMapping` can only work on the same kernel (and users need to manually disable some tiles?).

Reply:
Please check the updated README.md